identity/webapp/src/frontend/hooks/useUpdateUser.ts

```diff
@@ -24,43 +23,45 @@ type UseUpdateUserMutation = {
 };
 
 export function useUpdateUser(): UseUpdateUserMutation {
-  const { _updateUserState } = useUser();
+  const { reload: refreshUserSession } = useUser();
   const [state, setState] = useState<State>('initial');
   const [error, setError] = useState<UpdateUserError>();
 
-  const updateUser: UseUpdateUserMutation['updateUser'] = (
+  const updateUser: UseUpdateUserMutation['updateUser'] = async (
     userDetails,
     onComplete = () => void 0
   ) => {
     setState('loading');
-    callMiddlewareApi('PUT', '/account/api/users/me', userDetails)
-      .then(response => {
-        setState('success');
-        const updatedUser = response.data as UserInfo;
-        _updateUserState(updatedUser);
-        onComplete(updatedUser);
-      })
-      .catch((err: AxiosError) => {
-        setState('error');
-        switch (err.response?.status) {
-          case 401: {
-            setError(UpdateUserError.INCORRECT_PASSWORD);
-            break;
-          }
-          case 409: {
-            setError(UpdateUserError.EMAIL_ALREADY_EXISTS);
-            break;
-          }
-          case 429: {
-            setError(UpdateUserError.BRUTE_FORCE_BLOCKED);
-            break;
-          }
-          default: {
-            setError(UpdateUserError.UNKNOWN);
-            break;
-          }
+    try {
+      const updateResponse = await axios.put(
+        '/account/api/users/me',
+        userDetails
+      );
+      await refreshUserSession();
+      setState('success');
+      const updatedUser = updateResponse.data as UserInfo;
+      onComplete(updatedUser);
+    } catch (err) {
+      setState('error');
+      switch (err.response?.status) {
+        case 401: {
+          setError(UpdateUserError.INCORRECT_PASSWORD);
+          break;
         }
-      });
+        case 409: {
+          setError(UpdateUserError.EMAIL_ALREADY_EXISTS);
+          break;
+        }
+        case 429: {
+          setError(UpdateUserError.BRUTE_FORCE_BLOCKED);
+          break;
+        }
+        default: {
+          setError(UpdateUserError.UNKNOWN);
+          break;
```

Comment:
Minor: in other services we log the error when we hit the unknown error branch, because ideally we're never getting here. Worth a quick log here, or is it going to be logged elsewhere?

Reply:
Because this is frontend code I wouldn't usually bother logging it as you can see everything you might want to and more in your browser dev tools